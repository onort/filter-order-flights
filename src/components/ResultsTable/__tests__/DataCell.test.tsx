import React from "react"
import { shallow } from "enzyme"

import { DataCell } from "../"

describe("<DataCell />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<DataCell sub="Sub" val="Val" />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should render sub and value texts", () => {
    const wrapper = shallow(<DataCell sub="Sub" val="Val" />)
    expect(wrapper.find(".sub").text()).toBe("Sub")
    expect(wrapper.find(".val").text()).toBe("Val")
  })

  it("should have detailCell className if detailCell props is passed", () => {
    const wrapper = shallow(<DataCell sub="Sub" val="Val" detailCell={true} />)
    expect(wrapper.find(".detailCell").exists()).toBe(true)
  })

  it("should be able to have a custom className", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <DataCell className={testClassName} sub="Sub" val="Val" />
    )
    expect(wrapper.find(".dataCell").hasClass(testClassName)).toBe(true)
  })
})
