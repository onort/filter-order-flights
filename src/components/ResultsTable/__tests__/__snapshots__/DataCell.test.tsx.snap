// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataCell /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataCell
    detailCell={false}
    sub="Sub"
    val="Val"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span
          className="val"
        >
          Val
        </span>,
        <span
          className="sub"
        >
          Sub
        </span>,
        undefined,
      ],
      "className": "dataCell",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Val",
          "className": "val",
        },
        "ref": null,
        "rendered": "Val",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sub",
          "className": "sub",
        },
        "ref": null,
        "rendered": "Sub",
        "type": "span",
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span
            className="val"
          >
            Val
          </span>,
          <span
            className="sub"
          >
            Sub
          </span>,
          undefined,
        ],
        "className": "dataCell",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Val",
            "className": "val",
          },
          "ref": null,
          "rendered": "Val",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sub",
            "className": "sub",
          },
          "ref": null,
          "rendered": "Sub",
          "type": "span",
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<DataCell /> should match snapshot 2`] = `"<div class=\\"cell dataCell\\"><span class=\\"val\\">Val</span><span class=\\"sub\\">Sub</span></div>"`;
